src/main/routes/metrics-route.ts

```diff
@@ -1,72 +1,73 @@
 import { LensApiRequest } from "../router"
 import { LensApi } from "../lens-api"
-import { PrometheusClusterQuery, PrometheusIngressQuery, PrometheusNodeQuery, PrometheusPodQuery, PrometheusProvider, PrometheusPvcQuery, PrometheusQueryOpts } from "../prometheus/provider-registry"
+import { Cluster } from "../cluster"
+import _ from "lodash"
 
 export type IMetricsQuery = string | string[] | {
   [metricName: string]: string;
 }
 
+// This is used for backoff retry tracking.
+const MAX_ATTEMPTS = 5
+const ATTEMPTS = [...(_.fill(Array(MAX_ATTEMPTS - 1), false)), true]
+
+// prometheus metrics loader
+async function loadMetrics(promQueries: string[], cluster: Cluster, prometheusPath: string, queryParams: Record<string, string>): Promise<any[]> {
+  const queries = promQueries.map(p => p.trim())
+  const loaders = new Map<string, Promise<any>>()
+
+  async function loadMetric(query: string): Promise<any> {
+    async function loadMetricHelper(): Promise<any> {
+      for (const [attempt, lastAttempt] of ATTEMPTS.entries()) { // retry
+        try {
+          return await cluster.getMetrics(prometheusPath, { query, ...queryParams })
```

Comment:
You don't need to `await` here?

Reply:
You do because I want to catch the error if it happens. Normally just a `return` would be enough (and actually preferable because it is more performant) but since I want to intercept any rejected promises I need to do a `return await`.

Reply:
Ok, good to know